plugins/woocommerce-blocks/bin/add-split-chunk-dependencies.js

```diff
@@ -0,0 +1,68 @@
+// Dependency Extraction Webpack Plugin's asset.php array entries always refer to a single module with a single corresponding
+// js file. When we generate a split chunk it can represent multiple modules but it becomes an essential dependency
+// for any module in the bundle. To get around this limitation, we prefix the split chunk handles to the asset.php
+// file's dependency array. You must still register the handles for them in PHP for this to work but you won't have
+// to enqueue the split chunks per entry-point for example.
+class AddSplitChunkDependencies {
```

Comment:
This plugin will look through all generated assets with an asset.php file and add split chunks as dependencies to it. The name of the handle will be the configured name in Webpack. in PHP we will still need to register the vendor bundles but we won't have to enqueue them selectively for entry points that don't need them.

Reply:
I'd love an actual example of what we're solving here, as the text feels a bit too abstract for me.

Reply:
I've tried to improve the comment based on what I said in the above comment https://github.com/woocommerce/woocommerce/pull/48010/commits/b1b72e3ef5b358ce887aeba0ca2afb84da0f61d2